go-controller/pkg/node/management-port_linux.go

```diff
@@ -280,6 +293,108 @@ func createPlatformManagementPort(interfaceName string, localSubnets []*net.IPNe
 	return cfg, nil
 }
 
+func getIPTablesForHostSubnets(hostSubnets []*net.IPNet) (util.IPTablesHelper, util.IPTablesHelper, error) {
+	var ipt4, ipt6 util.IPTablesHelper
+	var err error
+
+	for _, hostSubnet := range hostSubnets {
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			if ipt6 != nil {
+				continue
+			}
+			ipt6, err = util.GetIPTablesHelper(iptables.ProtocolIPv6)
+		} else {
+			if ipt4 != nil {
+				continue
+			}
+			ipt4, err = util.GetIPTablesHelper(iptables.ProtocolIPv4)
+		}
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	return ipt4, ipt6, nil
+}
+
+func checkExistingPlatformMgmtPort(hostSubnets []*net.IPNet, mgmtPortName string) error {
+	_, stderr, err := util.RunOVSVsctl("list", "Interface", mgmtPortName)
+	if err != nil {
+		if !strings.Contains(stderr, "no row \""+mgmtPortName+"\" in table") {
+			return fmt.Errorf("failed to lookup OVS port: %s", stderr)
+		}
+		// Not found on the bridge. But could be that interface with the same name exists
+		return unconfigurePlatformMgmtInterface(hostSubnets, mgmtPortName)
+	}
+
+	// Found existing port. Check its type
+	stdout, stderr, err := util.RunOVSVsctl("get", "Interface", mgmtPortName, "type")
+	if err != nil {
+		return fmt.Errorf("failed to get OVS interface type: %s", stderr)
+	}
+	if stdout == "internal" {
+		klog.Infof("Found OVS internal port. Removing it")
+		_, stderr, err := util.RunOVSVsctl("del-port", "br-int", mgmtPortName)
+		if err != nil {
+			return fmt.Errorf("failed to remove OVS internal port: %s", stderr)
+		}
+		return nil
+	}
+
+	// It is other representor which was used as management port.
+	// Remove it from the bridge and rename.
+	klog.Infof("Found existing representor management port. Removing it")
+	return unconfigureMgmtRepresentorPort(mgmtPortName)
+}
+
+func unconfigureMgmtRepresentorPort(mgmtPortName string) error {
+	_, stderr, err := util.RunOVSVsctl("del-port", "br-int", mgmtPortName)
```

Comment:
Add `--if-exists` in the ovsctl run?

Reply:
Agree, it is not strictly needed here.

Reply:
@DmytroLinkin please add the argument `--if-exists`.
This would be good just as a safety measure if someone outside of the scope of ovn-kubernetes makes any operations to that port.